scripts/stackstorm.js

```diff
@@ -283,6 +293,12 @@ module.exports = function(robot) {
       } else {
         data = req.body;
       }
+      // Special handler to try and figure out when a hipchat message
+      // is a whisper:
+      if (robot.adapterName == 'hipchat' && !data.whisper && data.channel.indexOf('@') > -1 ) {
```

Comment:
`@` in room names are ignored when getting the full ID. I've set up `bottesting` and `bot@testing` channels, Hubot shows them as `99273_bottesting` and `99273_bottesting_1`. So your test is just fine.